arrays/arrays.c

```diff
@@ -47,13 +52,21 @@ void destroy_array(Array *arr) {
 void resize_array(Array *arr) {
 
   // Create a new element storage with double capacity
+  int new_storage = 2 * arr->capacity;
+  char ** new_elements = calloc(new_storage, sizeof(char *));
+  //char ** new_elements = malloc(sizeof(char *));
 
   // Copy elements into the new storage
+  for (int i = 0; i < arr->capacity; i++) {
```

Comment:
Should only be copying over elements where they exist. That means you should use arr->count instead of arr->capacity. It's not wrong but you may end up copying over NULL data. It's not necessary.